Update SpawnPointBehavior.cs / InputManager.cs

maximalum opened the pull request:
Possibly testing fade's buttons.. But in all honesty--testing styles of code and commenting that allow new contributors to understand what the H is going on.  [My favorite Coding Convention & Standards resource](https://en.wikibooks.org/wiki/C_Sharp_Programming/Naming). Naming Documentation from [Microsoft](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions). More [Naming Conventions & Guidelines](https://csharpcodingguidelines.com/naming-guidelines/) but focus more on *why?*

maximalum committed:
Update SpawnPointBehavior.cs

Possibly testing fade's buttons.. But in all honesty--testing styles of code and commenting that allow new contributors to understand what the H is going on.  One resource of many: https://en.wikibooks.org/wiki/C_Sharp_Programming/Naming

pvutov commented on src/FieldWarning/Assets/Ingame/UI/SpawnPointBehaviour.cs:
```diff
@@ -11,55 +11,46 @@
+        // Increase spawn queue time if total amount in queue is greater than zero
+        //  otherwise set to QUEUE_DELAY (1f) PS: Turnary > if statements
+        this._spawnTime = (this.spawnQueue.Count > 0) ?
+            this._spawnTime += MIN_SPAWN_INTERVAL : QUEUE_DELAY;
```
I find this construction a lot less readable. I would avoid the ternary operator when the two outcomes are not simple assignments.  If you need a comment to explain what your code is doing, maybe the code a bit too cleverly written?

maximalum replied:
I can see why you'd think that but, my view, makes it show it's increasing `_spawnTime` w/ `+=`. rather than setting it to a constant. A long with comments. Any input on why no do or pursue?

pvutov replied:
This is one of  the simplest possible constructs and the proposed change makes it hard to read. This should not require a comment, the original form is a block that takes 2 seconds to understand.

maximalum replied:
> 
> 
> This is one of the simplest possible constructs and the proposed change makes it hard to read. This should not require a comment, the original form is a block that takes 2 seconds to understand.

There was already a comment there. I added a small bit with humor intended (sorry it wasn't a good German joke lmao). This kind of `construct` is used in a vast amount of projects. It's nothing new, not over complicated, and its serving its purpose.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/SpawnPointBehaviour.cs:
```diff
@@ -11,55 +11,46 @@
-    private Vector3 oldPosition;
-    private Queue<GhostPlatoonBehaviour> spawnQueue = new Queue<GhostPlatoonBehaviour>();
-    private float spawnTime = MIN_SPAWN_INTERVAL;
+    private float _spawnTime = MIN_SPAWN_INTERVAL;
```
Nice style cleanup but you missed the name fix for spawnQueue :+1:

maximalum replied:
That's what I get for trusting VSC linter. :'[

maximalum committed:
Fixed SpawnPointBehaviour.cs Refactoring

Renamed `spawnQueue` to `_spawnQueue` and added missing `this` statements. Thanks fade.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/SpawnPointBehaviour.cs:
```diff
@@ -11,55 +11,46 @@
+        if (!this.spawnQueue.Any()) return;
+
+        // Check if spawn time minus the current deltaTime is > 0 then exit method;
+        if (this._spawnTime -= Time.deltaTime > 0) return;
```
This code is self-explanatory and does not need comments. The return should be on its own line, this inlining makes it hard to skim for no reason.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/SpawnPointBehaviour.cs:
```diff
@@ -11,55 +11,44 @@
-    {
-        ghostPlatoons.ForEach(x => spawnQueue.Enqueue(x));
+    public void BuyPlatoons(List<GhostPlatoonBehaviour> ghostPlatoons) {
+        ghostPlatoons.ForEach (x => this._spawnQueue.Enqueue(x));
```
Against the style guide. If you are proposing an intentional change then lets talk on discord.

maximalum replied:
I personally didn't touch that. I'm sure ReSharper did. But yes, I'd like to chat with you about some changes to the style guide.

maximalum committed:
Refactoring and Optimization of InputManager.cs

This was a test-run but more of a trial. What could I do? How could I push it? How much was impractical or poor practice? What are the good things? That's what I'm trying to get out of this. Examples: `Commands` are now lambda expressions, narrowed down the if/switch, added different exceptions, utilized renaming, removed different functions, etcetra. Input, suggestions, "holy shit that sucks" woiuld all be nice. This isn't close to a final project and wasn't even aimed towards the *official* Coding Style.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+                    break;
+
+                case Modes.FirePosition:
+                    if (this.IsMouse(Events.Down))
```
This takes a method that any Unity developer will recognize and effectively renames it. Not worth it.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+                    this._selectionManager.DispatchMoveCommand(false, MoveWaypoint.MoveMode.Fast);
+
+                    if (this.IsMouse(Events.Down) && !KeyDown(KeyCode.LeftShift) || this.IsMouse(Events.Down, 1))
+                        this.SetCursor(Modes.Normal);
```
I was originally of the opinion that `this` can be used as often as one wants but this is excessive. If I had to make a choice I would prefer to ban `this` completely.

gatchrat-zz replied:
i dont really see much advantage for using it,since the code shouldnt contain two variables with the same name and therefore using "this" should never be necessary. I would like to know why you used it this much, since it seems to just increase the length of the code and decrease the readability.

maximalum replied:
There are certain areas of code I would look at (no, I can't think of anything off the top of my head) and be genuinely confused as to what the hell was what. The reason `this` is used so much is because of me sticking to a convention and using a tool to automatically do so. If a single word obstructs your vision of the code enough to mess with readability, I think you might need glasses. haha.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+            Purchasing,
+            FirePosition,
+            ReverseMove,
+            FastMove
```
Enum values should be UPPERCASE as any constant. Not that the original code was correct, but since you're renaming them anyway..

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -11,17 +11,19 @@
+
+    using UnityEngine;
+    using UnityEngine.EventSystems;
+
```
The entire codebase puts includes outside the namespace, and most incoming programmers will also be used to that ordering. Since both approaches are largely equivalent, I think it's easier for you to change your stylecop defaults or whatever is pushing you to use this style.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
         }
 
-        void Update()
+        public enum Modes
```
Plural in an enum seems unnatural to me and SO agrees:
https://stackoverflow.com/questions/15755955/naming-of-enums-in-java-singular-or-plural
https://stackoverflow.com/questions/1405851/enum-naming-convention-plural

I would also really prefer a more expressive name. What modes are these? Firing modes? Launch modes? I don't understand why we're trying to save 5 characters here.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
 
-            MaybePurchaseGhostUnits(closestSpawn);
+            if (!this._spawnPointList.Contains(monoBehavior))
+                this._spawnPointList.Add(monoBehavior);
```
`monoBehavior` is an unfortunate name because it makes it seem like a unity inbuilt (like `transform`). `spawn`? `spawnPoint`?

gatchrat-zz replied:
Agree,maybe rename it to spawnPointBehavior ? Or is that against convention ?

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
         }
 
-        void OnOrderLongClick()
+        public void OnGui()
```
This can be private as you did with Update().

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
 
-        // Show a Symbol at the position where a move order was issued:
-        private void DisplayOrderFeedback()
+        public void RegisterActor(MonoBehaviour monoBehaviour, bool birth = true)
```
We already talked about this when you wanted to do it in MatchSession.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
         }
 
-        public void OnGUI()
+        private PlayerData LocalPlayer => this.Session.LocalPlayer.Data;
```
This is pretty neat and an interesting case of public/private naming. I guess keeping it private but named as a public does make the most sense.

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
         }
 
-        private void EnterFastMoveMode()
+        private void SetCursor(Modes modes)
```
This method does not just set the cursor. The old name of "Enter___Mode" reflected that. Also your enum naming confused you here, the `modes` variable actually only holds one mode at a time :D

pvutov commented on src/FieldWarning/Assets/Ingame/UI/InputManager.cs:
```diff
@@ -30,388 +32,327 @@ namespace PFW.Ingame.UI
+                default: throw new ArgumentOutOfRangeException(nameof(modes), modes, null);
+            }
+
+            Cursor.SetCursor(nextMouseTexture, nextMouseLocation, CursorMode.Auto);
```
I find this less readable. With four methods you can have it stated black on white: `Cursor.SetCursor(_primedReticle, Vector2.zero)`. Now I have to read this switch statement that changes a different variable in each clause? What are we gaining here?

maximalum replied:
I went with what was there and tried to modify things for *ideas* and **suggestions**. I did my best to avoid deleting entire snippets.
